lib/onfleet/resource.rb

```diff
@@ -96,13 +99,85 @@ def self.delete(action = "", values = {})
     def self.request(uri, options = {})
       options = Nestful::Helpers.deep_merge(self.options, options)
 
-      response = Nestful::Request.new(
-        uri, options
-      ).execute
+      begin
+        response = Nestful::Request.new(
+          uri, options
+        ).execute
+      rescue JSON::ParserError
+        raise Errors::Error
+      rescue Nestful::ResponseError => e
```

Comment:
I'd rescue all `Nestful::Error` instances here (or in a diff rescue clause), as it includes some [other weird connection issues](https://github.com/maccman/nestful/blob/master/lib/nestful/exceptions.rb#L11) that might show up.